src/v/coproc/event_listener.cc

```diff
@@ -48,90 +46,54 @@ ss::future<> event_listener::stop() {
-              vlog(
-                coproclog.info,
-                "Moving coprocessor {} to inactive dir",
-                active_path);
-              return ss::rename_file(active_path.string(), mvto.string());
-          } else {
-              if (exists) {
-                  vlog(
-                    coproclog.warn,
-                    "Attempt to deploy coprocessor with script {} but one "
-                    "already exists: {}",
-                    name,
-                    active_path);
-                  return ss::now();
-              }
-              vlog(
-                coproclog.info,
-                "Writing coprocessor {} to submit dir {}",
-                name,
-                _submit_dir);
-              std::filesystem::path submit_src(_submit_dir / name.c_str());
-              return write_fully(submit_src, std::move(source_code));
-          }
-      });
+event_listener::persist_actions(absl::btree_map<script_id, iobuf> wsas) {
+    std::vector<enable_copros_request::data> enables;
+    std::vector<script_id> disables;
+    for (auto& [id, source] : wsas) {
+        /// Keeping the active_ids cache up to date solves the issue of issuing
+        /// a bunch of remove commands for scripts that aren't deployed (this
+        /// could occur during bootstrapping)
+        auto found = _active_ids.find(id);
+        if (query_action(source) == event_action::remove) {
+            if (found != _active_ids.end()) {
+                _active_ids.erase(found);
+                disables.emplace_back(id);
+            }
+        } else {
+            /// ... or in the case if deploys are performed using the same key.
+            /// This would normally be resolved if the events came in the same
+            /// record batch by the reconcile events function, but not if they
+            /// arrived in separate batches.
+            if (found == _active_ids.end()) {
+                _active_ids.insert(id);
+                enables.emplace_back(enable_copros_request::data{
+                  .id = id, .source_code = std::move(source)});
+            }
+        }
+    }
+    if (!enables.empty()) {
+        enable_copros_request req{.inputs = std::move(enables)};
+        co_await _dispatcher.enable_coprocessors(std::move(req));
+    }
+    if (!disables.empty()) {
+        disable_copros_request req{.ids = std::move(disables)};
+        co_await _dispatcher.disable_coprocessors(std::move(req));
+    }
 }
 
-event_listener::event_listener(std::filesystem::path data_dir)
-  : _client(
-    std::vector<unresolved_address>{unresolved_address("127.0.0.1", 9092)})
-  , _wasm_root(data_dir.parent_path() / "coprocessor")
-  , _active_dir(_wasm_root / "active")
-  , _submit_dir(_wasm_root / "submit")
-  , _inactive_dir(_wasm_root / "inactive") {}
+event_listener::event_listener(ss::sharded<pacemaker>& pacemaker)
+  : _client({config::shard_local_cfg().kafka_api()[0].address})
```

Comment:
does it need to be documented some where that the first entry in kafka_api config affects co-proc client behavior?

Reply:
Yeah sure, where do you think the best place to put a comment like that would be?